### PR TITLE
DAOS-4927 pydaos: Remove dbm code and fold kv_by_name into pydaos.

### PR DESCRIPTION
**Pull request by ashleypittman**
Remove some code which was intended as a POC, keeping only
the get_kv_by_name() method which is moved into the Cont
object and cleaned up to allow alternative roots and opening
but not creating kvs.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/client/pydaos/pydaos_core.py [hunk 1]
```diff
@@ -154,6 +159,30 @@ def rootkv(self, cid=ObjClassID.OC_SX):
...
+        if name in root:
+            object_data = pickle.loads(root[name])
+            return self.kv(object_data['oid'])
+
+        if not create:
+            raise KvNotFound
+
+        new_kv = self.newkv()
+        # Create a new entry in the root kv, where the entry
+        # itself is a dick, and the 'oid' entry is the object
```
Comment (mjmac) [inline src/client/pydaos/pydaos_core.py:179]: I think you've got a typo here where you meant 'dict'...

Comment (ashleypittman) [reply src/client/pydaos/pydaos_core.py:179]: I think I probably did.

File: src/client/pydaos/pydaos_core.py [hunk 2]
```diff
@@ -154,6 +159,30 @@ def rootkv(self, cid=ObjClassID.OC_SX):
...
+        optionally create kv if not found"""
+
+        if not root:
+            root = self.rootkv()
+        if name in root:
+            object_data = pickle.loads(root[name])
+            return self.kv(object_data['oid'])
+
+        if not create:
+            raise KvNotFound
```
Comment (phender) [inline src/client/pydaos/pydaos_core.py:175]: Should this error indicate which `name` was not found when not creating?

File: src/client/pydaos/pydaos_core.py (hunk 1 above)
Comment (johannlombardi) [reply src/client/pydaos/pydaos_core.py:179]: :)

File: src/client/pydaos/pydaos_core.py (hunk 2 above)
Comment (ashleypittman) [reply src/client/pydaos/pydaos_core.py]: Ok, I added this which works, and I added a test for it.

It is causing some problems however, it seems that when I do that there's an extra reference taken on the container somehow so the node_local_test code doesn't try to close the container until after it's stopped the DAOS server.  This wouldn't be an issue in normal use as the daos server wouldn't be a child of the python process but it does make it difficult to add a test case for it using the current technique.

**Pull request merged**